### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by schaefert1**
## Checklist
- [ ] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by xindizhang — APPROVED** [review]
**Great work — no further changes needed!** 

Well-written and well-organized pipeline! Excellent use of training and testing data: using training data for training, hyperparameter tuning, and model selection, and reserving the test set for final evaluation. Your feature interpretation is also very strong.

Just a couple comments:
- including a brief PR description is good practice, as it helps your "project-mates" quickly understand what’s going on. 
- Ecellent skewness check before preprocessing 2! You could also consider exploring descriptive statistics and visualizing feature distributions when making preprocessing decisions.
- You used joblib to export the model, which is different from what the assignment specified (pickle). That’s totally fine — I’ll accept it — but make sure you also understand how to use pickle, as it's good to be familiar with both.

**Comment by xindizhang** [comment]
Providing visualizations with SHAP values (beeswarm and waterfall plots) would also be a great addition. They make it much more straightforward for the reader and yourself to understand and interpret how the model is making decisions.
